trax/tf_numpy/examples/mnist/dataset.py

```diff
@@ -13,77 +13,76 @@
-
-  return (shuffle(training_data), validation_data, test_data)
-
-
-def to_one_hot(label, num_classes=10):
-  vec = np.zeros(num_classes, dtype=np.float32)
-  vec[label] = 1.
-  return vec
-
-
-def _maybe_download():
-  """Downloads the MNIST dataset if it is not there already."""
-  data_url = 'http://www.iro.umontreal.ca/~lisa/deep/data/mnist/mnist.pkl.gz'
-  filename = data_url.split('/')[-1]
-  filepath = os.path.join(_get_data_dir(), filename)
-  if not os.path.exists(filepath):
-
-    def _progress(count, block_size, total_size):
-      print('\r>> Downloading %s %.1f%%' %
-            (filename, float(count * block_size) / float(total_size) * 100.0))
-
-    filepath, _ = urllib.urlretrieve(data_url, filepath, _progress)
-    statinfo = os.stat(filepath)
-    print('Successfully downloaded %s %d bytes.' % (filename, statinfo.st_size))
-  else:
-    print('Data already present on disk.')
-  return filepath
-
-
-def _get_data_dir():
-  return '/tmp'
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow.python.ops import numpy_ops as np
+
+# current convention is to import original numpy as np
+import numpy as onp
+
+
+class MNIST():
+    "This module tries to mimmick the NumPy Keras Dataset"
+    def __init__(self, batch_size):
+        """Initializes the MNIST dataset """
+        self.batch_size = batch_size
+
+        print('Loading data')
+        mnist_train = tfds.load('mnist', split='train', shuffle_files=True, batch_size=-1)
+        mnist_test = tfds.load('mnist', split='test', shuffle_files=True, batch_size=-1)
+
+
+        self.mnist_train_images, self.mnist_train_labels = mnist_train['image'], mnist_train['label']
+        self.mnist_test_images, self.mnist_test_labels = mnist_test['image'], mnist_test['label']
+
+        self.train_len = self.mnist_train_images.shape[0]
+        self.val_len = self.mnist_test_images.shape[0]
+
+        print('Loaded dataset with {} training and {} test examples.'.format(
+                self.mnist_train_images.shape[0], self.mnist_test_images.shape[0]))
+
+        assert(self.mnist_train_images.shape[0] == self.mnist_train_labels.shape[0])
+        assert(self.mnist_test_images.shape[0] == self.mnist_test_labels.shape[0])
+        assert(self.mnist_train_images.shape[0] % batch_size == 0)
+        assert(self.mnist_test_images.shape[0] % batch_size == 0)
+
+
+    def iterator(self, split, infinite=True):
+        """Build the iterator for inputs."""
+        if split=='train':
+            mnist_images, mnist_labels = self.mnist_train_images, self.mnist_train_labels
+        elif split=='test':
+            mnist_images, mnist_labels = self.mnist_test_images, self.mnist_test_labels
+
+        len_dim_0 = mnist_images.shape[0]
+        mnist_images = np.reshape(mnist_images, (len_dim_0, 784))
+        mnist_labels = np.asarray(tf.one_hot(mnist_labels, 10))
+
+        mnist_images, mnist_labels = self.shuffle(mnist_images, mnist_labels)
+
+        index = 0
```

Comment:
Is it possible to have the tfds.load(..., batch_size=self.batch_size)?

Alternatively, the loaded dataset can be batched directly? https://www.tensorflow.org/api_docs/python/tf/data/Dataset#batch

I see that this is yielding actual training data, it'd be nice to instead return an actual iterator on the underlying dataset.